src/components/file-browser/FileBrowserDirectories.ts

```diff
@@ -0,0 +1,369 @@
+        role: 'treeitem',
+        style: {
+          display: 'flex',
+          flexDirection: 'column'
+        }
+      }, [
+        h(Link, {
+          role: 'presentation',
+          tabIndex: -1,
+          style: {
+            // Align with directory tree item arrow icons.
+            padding: `0.25rem 0.5rem 0.25rem ${level + 1.25}rem`,
+            borderColor: `${parentId}-load-next` === activeDescendant ? colors.accent() : 'transparent',
+            borderStyle: 'solid',
+            borderWidth: '1px 0'
+          },
+          onClick: () => loadNextPage()
+        }, ['Load next page'])
+      ])
+    ])
+  ])
+}
+
+interface FileBrowserDirectoryProps {
+  activeDescendant: string
+  id: string
+  provider: FileBrowserProvider
+  level: number
+  path: string
+  rootLabel: string
+  selectedDirectory: string
+  setActiveDescendant: Dispatch<SetStateAction<string>>
+  onSelectDirectory: (path: string) => void
+}
+
+export const FileBrowserDirectory = (props: FileBrowserDirectoryProps) => {
+  const {
+    activeDescendant,
+    id,
+    level,
+    path,
+    provider,
+    rootLabel,
+    selectedDirectory,
+    setActiveDescendant,
+    onSelectDirectory
+  } = props
+  const isSelected = path === selectedDirectory
+
+  const [isExpanded, setIsExpanded] = useState(false)
+  const [hasLoadedContents, setHasLoadedContents] = useState(false)
+
+  return li({
+    'aria-expanded': isExpanded,
+    // Label with the link to read only the directory basename instead of both the basename and the subdirectories list label.
+    'aria-labelledby': `${id}-link`,
+    // aria-level starts at 1, level starts at 0.
+    'aria-level': level + 1,
+    // aria-selected: false results in every tree item being read as "selected".
+    'aria-selected': isSelected ? true : undefined,
+    // Data attribute allows getting the directory path from the activedescendant element ID.
+    'data-path': path,
+    id,
+    role: 'treeitem',
+    style: {
+      display: 'flex',
+      flexDirection: 'column',
+      position: 'relative'
+    }
+  }, [
+    // Wrapper span provides a larger click target than just the icon.
+    span({
+      'aria-hidden': true,
+      'data-testid': 'toggle-expanded',
+      style: {
+        position: 'absolute',
+        top: '1px',
+        left: `${level - 1}rem`,
+        display: 'flex',
+        justifyContent: 'flex-end',
+        alignItems: 'center',
+        width: '2rem',
+        height: '1.5rem'
+      },
+      onClick: () => {
+        setIsExpanded(!isExpanded)
+        if (isExpanded && activeDescendant.startsWith(`${id}-`)) {
+          setActiveDescendant(id)
+        }
+      }
+    }, [
+      icon(Utils.cond(
+        [isExpanded && hasLoadedContents, () => 'angle-down'],
+        [isExpanded && !hasLoadedContents, () => 'loadingSpinner'],
+        [!isExpanded, () => 'angle-right']
+      ), {
+        // @ts-expect-error
+        color: isSelected ? '#000' : colors.accent(),
```

Comment:
It seems like we should define `#000` as a variable (at least within this file).

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/3469/commits/9ddd07e5eacaf47087d534a45b20cdc276230049.